qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -287,25 +313,42 @@ def run(self, run_input: Union[QuantumCircuit, List[QuantumCircuit]], **run_opti
         if isinstance(run_input, QuantumCircuit):
             run_input = [run_input]
 
-        experiment_unitaries = copy.deepcopy(self.default_pulse_unitaries)
-
         for circuit in run_input:
+            unitaries = {}
+
+            # 1. Parse the calibrations and simulate any new schedule. Add U to the unitaries.
             for name, schedule in circuit.calibrations.items():
                 for (qubits, params), schedule_block in schedule.items():
-                    if (name, qubits, params) not in experiment_unitaries:
-                        experiment_unitaries[(name, qubits, params)] = self.solve(
-                            schedule_block, qubits
-                        )
+                    schedule_key = hash(repr(schedule))
 
-            state_t = self.gound_state.copy()
+                    # Simulate the schedule if not in the cache.
+                    if schedule_key not in self._schedule_cache:
```

Comment:
the `repr` of `schedule` does not contain qubit index info. This could be a problem when running experiments with multiple qubits since they could have different qubit parameters such as frequency. Or are we planning for them to be identical?

Reply:
I think this should be fine. Indeed the repr of the schedule does contain the qubit index. It is in the drive channel. E.g.
```python
with pulse.build() as sched:
    pulse.play(pulse.Drag(160, 0.5, 40, -0.34), pulse.DriveChannel(2))

repr(sched)
```
prints
```
'ScheduleBlock(Play(Drag(duration=160, amp=(0.5+0j), sigma=40, beta=-0.34), DriveChannel(2)), name="block6", transform=AlignLeft())'
```
Here you can see `DriveChannel(2)` which will change depending on the qubit.